discovery.md

```diff
@@ -0,0 +1,378 @@
+# CloudSubscriptions: Discovery - Version 0.1
+
+## Abstract
+
+CloudSubscriptions Discovery API is a vendor-neutral API specification for
+determining what events an Event Producer or middleware (Aggregator, Broker)
+has available, as well as how to subscribe to those events.
+
+## Status of this document
+
+This is a working draft. Breaking changes could be made in the next minor
+version.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Notations and Terminology](#notations-and-terminology)
+- [API Specification](#api-specification)
+- [Protocol Bindings](#protocol-bindings)
+- [Privacy & Security](#privacy-and-security)
+
+## Overview
+
+The goal of the CloudDiscovery API is to enable connection between consumers of
+events and producers / aggregators / brokers of events and to facilitate the
+creation of CloudSubscriptions. The discovery and subscription APIs are
+control-plane APIs to enable events to be delivered over the eventing
+data-plane.
+
+Discovery must allow for an event producer or middleware component to advertise
```

Comment:
we'll need to add the RFC2119 boilerplate text and go thru and verify each use of the keywords - but not needed for the first PR. Just FYI

Reply:
ack. I did remove all uses of "middleware"